### ISSUE TITLE
Relation picker issue

### ISSUE THREAD

**Issue by galash13** [pocketbase/pocketbase#2675]
hello, i have 2 collections that are related to each other (one to many). when i try to insert a record in the second section and click the relation picker button the app crashes with this error in the console:

`e.substring is not a function`

after i added the item from a postman call the list of the second collection also fails to load and the app crashes due to the same error.

both collections have a field of type JSON, and the second collection also has 1 more string field and the relation field with the first collection

issue present on the latest version of pocketbase 0.16.4

<img width="449" alt="Screenshot 2023-06-08 at 10 26 45" src="https://github.com/pocketbase/pocketbase/assets/56839513/46ed251a-cbb2-4005-addc-17f27af2e6a7">
<img width="666" alt="Screenshot 2023-06-08 at 10 31 16" src="https://github.com/pocketbase/pocketbase/assets/56839513/ed89fae9-4cdb-4662-9718-81b7dcf8eb81">
<img width="842" alt="Screenshot 2023-06-08 at 10 27 44" src="https://github.com/pocketbase/pocketbase/assets/56839513/4b03aa48-79d9-4ec5-8cb3-16b79b827131">

**Comment by ganigeorgiev** [comment]
Could you export your collections configuration as JSON (_Admin UI > Settings > Export collections_)?

**Comment by ganigeorgiev** [comment]
I think I found how to reproduce it. The `label` json field is causing the error because it is used as "Display field" but it is not a string.
I'll submit a fix sometime later today, but as a temp workaround you can try to explicitly set another "Display field" from the `relation` field options (eg. the `id`).

**Comment by galash13** [comment]
> Could you export your collections configuration as JSON (_Admin UI > Settings > Export collections_)?

```
[
    {
        "id": "np1wu6sn67l2olq",
        "name": "sections",
        "type": "base",
        "system": false,
        "schema": [
            {
                "id": "qhbhmyiq",
                "name": "title",
                "type": "json",
                "system": false,
                "required": true,
                "options": {}
            }
        ],
        "indexes": [
            "CREATE INDEX `_np1wu6sn67l2olq_created_idx` ON `sections` (`created`)"
        ],
        "listRule": "",
        "viewRule": null,
        "createRule": "",
        "updateRule": "",
        "deleteRule": null,
        "options": {}
    },
    {
        "id": "fzxe3t8zwemxnng",
        "name": "links",
        "type": "base",
        "system": false,
        "schema": [
            {
                "id": "zsiitzt3",
                "name": "label",
                "type": "json",
                "system": false,
                "required": true,
                "options": {}
            },
            {
                "id": "rmvm4t2f",
                "name": "link",
                "type": "text",
                "system": false,
                "required": true,
                "options": {
                    "min": null,
                    "max": null,
                    "pattern": ""
                }
            },
            {
                "id": "jhpafc3m",
                "name": "section",
                "type": "relation",
                "system": false,
                "required": true,
                "options": {
                    "collectionId": "np1wu6sn67l2olq",
                    "cascadeDelete": true,
                    "minSelect": null,
                    "maxSelect": 1,
                    "displayFields": []
                }
            }
        ],
        "indexes": [
            "CREATE INDEX `_fzxe3t8zwemxnng_created_idx` ON `links` (`created`)"
        ],
        "listRule": "",
        "viewRule": null,
        "createRule": "",
        "updateRule": "",
        "deleteRule": null,
        "options": {}
    }
]
```

**Comment by galash13** [comment]
> I think I found how to reproduce it. The `label` json field is causing the error because it is used as "Display field" but it is not a string. I'll submit a fix sometime later today, but as a temp workaround you can try to explicitly set another "Display field" from the `relation` field options (eg. the `id`).

can't seem to find this option, in the edit collection panel i can only change the nonempty toggle of my field

**Comment by ganigeorgiev** [comment]
It should be fixed in the latest [v0.16.5 release](https://github.com/pocketbase/pocketbase/releases/tag/v0.16.5).
